akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -712,14 +713,19 @@ object Http extends ExtensionId[HttpExt] with ExtensionIdProvider {
    * Represents a connection pool to a specific target host and pool configuration.
    */
   final case class HostConnectionPool(setup: HostConnectionPoolSetup)(
-    private[http] val gatewayFuture: Future[PoolGateway]) extends javadsl.HostConnectionPool { // enable test access
+    private[http] val gatewayFuture: Future[PoolGateway]) { // enable test access
 
     /**
      * Asynchronously triggers the shutdown of the host connection pool.
      *
      * The produced [[Future]] is fulfilled when the shutdown has been completed.
      */
-    def shutdown()(implicit ec: ExecutionContext): Future[Unit] = gatewayFuture.flatMap(_.shutdown())
+    def shutdown()(implicit ec: ExecutionContextExecutor): Future[Done] = gatewayFuture.flatMap(_.shutdown())
+
+    private[http] def toJava = new akka.http.javadsl.HostConnectionPool {
+      override def setup = HostConnectionPool.this.setup
+      override def shutdown(executor: ExecutionContextExecutor): CompletionStage[Done] = HostConnectionPool.this.shutdown()(executor).toJava
```

Comment:
So that the Java can pass an executor easier, I think at least, following @rkuhn Java 8 API changes here.